packages/project-access/src/file/file-search.ts

```diff
@@ -0,0 +1,91 @@
+import type { Editor } from 'mem-fs-editor';
+import { basename, dirname, join, sep } from 'path';
+import { default as find } from 'findit2';
+import { fileExists } from './file-access';
+
+/**
+ * Add missing dump properties
+ */
+declare module 'mem-fs-editor' {
+    type FileMap = { [key: string]: { state: 'modified' | 'deleted' } };
+
+    export interface Editor {
+        dump(cwd: string): FileMap;
+    }
+}
+
+/**
+ * Search for 'filename' starting from 'root' in the list of virtual (not committed) files. Returns array of paths that contain the file.
+ *
+ * @param files - - array of paths that contain the filename found on the filesystem
+ * @param filename - filename to search
+ * @param root - root folder to start search
+ * @param stopFolders - list of folder names to exclude (search doesn't traverse into these folders)
+ * @param fs - mem-fs-editor instance
+ * @returns - enhanced array of paths that contain the filename
+ */
+function checkVirtualFiles(files: string[], filename: string, root: string, stopFolders: string[], fs: Editor) {
```

Comment:
Consider converting the parameters to a destructured object? This function accepts a lot of params.

Does `exclude` sound better than `stopFolders`?

Reply:
changed `stopFolders` to `excludeFolders`. 
@tobiasqueck, any comments on the other points?

Reply:
What!? This function is a piece of art! Joke aside, I will rework it based on the very good feedback.

Reply:
> Instead of modifying the original array being passed in, does returning a new array make more sense? The caller can choose to extend the original array if they want

The incoming array is not being modified in the current code.

Reply:
Anyways, please review https://github.com/SAP/open-ux-tools/pull/685/commits/a4918f6c32f5b5f2685a385891642baf1ec09f32. I have split the functionality into two functions with a single purpose each. This also reduced the amount of parameters.